pkg/osquery/table/touchid_system_darwin.go

```diff
@@ -0,0 +1,88 @@
+package table
+
+import (
+	"bytes"
+	"context"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/go-kit/kit/log"
+	"github.com/pkg/errors"
+
+	"github.com/kolide/osquery-go"
+	"github.com/kolide/osquery-go/plugin/table"
+)
+
+func TouchIDSystemConfig(client *osquery.ExtensionManagerClient, logger log.Logger) *table.Plugin {
+	t := &touchIDSystemConfigTable{
+		client: client,
+		logger: logger,
+	}
+	columns := []table.ColumnDefinition{
+		table.IntegerColumn("touchid_compatible"),
+		table.TextColumn("secure_enclave_cpu"),
+		table.IntegerColumn("touchid_enabled"),
+		table.IntegerColumn("touchid_unlock"),
+	}
+
+	return table.NewPlugin("kolide_touchid_system_config", columns, t.generate)
+}
+
+type touchIDSystemConfigTable struct {
+	client *osquery.ExtensionManagerClient
+	logger log.Logger
+	config *touchIDSystemConfig
+}
+
+type touchIDSystemConfig struct {
+	touchIDCompatible int
+	secureEnclaveCPU  string
+	touchIDEnabled    int
+	touchIDUnlock     int
+}
+
+// TouchIDSystemConfigGenerate will be called whenever the table is queried.
+func (t *touchIDSystemConfigTable) generate(ctx context.Context, queryContext table.QueryContext) ([]map[string]string, error) {
+	var results []map[string]string
+	var touchIDCompatible, secureEnclaveCPU, touchIDEnabled, touchIDUnlock string
+
+	// Read the security chip from system_profiler
```

Comment:
The system_profiler stuff should never change. So it could be moved up into TouchIDSystemConfig. (I'll accept this as-is, and am willing to do that in future work)

Reply:
I'd like to look into this before we merge. Please hold off on merging for now.